### PR TITLE
Create `rpl_master_retry_count.test`

### PR DESCRIPTION
**Pull request by ParadoxV5**
* [x] ~~*The Jira issue number for this PR is: MDEV-35304*~~

## Description
I created a MTR test to test `--master-retry-count` (and also `master_connect_retry` as a dependency) because I suspect its `0` edge case doesn’t match [the KB description](https://mariadb.com/kb/en/mariadbd-options/#-master-retry-count).
My local build doesn’t have `debug_sync` so I’m borrowing our buildbot.

## Release Notes
N/A (if the edge case I suspected doesn’t apply)

## How can this PR be tested?
Neither `master_retry_count` ([MDEV-25674](https://jira.mariadb.org/browse/MDEV-25674)) nor `master_connect_retry` are global variables, so I had to hardcode them.

## Basing the PR against the correct MariaDB version
* [ ] *This is a new feature or a refactoring, and the PR is based against the `main` branch.*
* [ ] *This is a bug fix, and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

## PR quality check
* [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/-/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
* [x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

File: mysql-test/suite/rpl/t/rpl_master_retry_count.test [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    LET $rpl_connection_name= server_`i`;
+    SOURCE include/rpl_connection.inc;
+    SET master_connect_retry= (i-1) % 3; # 1, 2, 0
+
+    ECHO Case 1: Retry for reconnecting from a simulated disconnection;
+    # This section is based on include/rpl_get_master_version_and_clock.test
+
+    SET original_dbug = @@GLOBAL.debug_dbug;
+    SET @@GLOBAL.debug_dbug= "d,debug_lock.before_get_UNIX_TIMESTAMP";
+    CHANGE PRIMARY TO master_connect_retry= master_connect_retry;
```
Comment (ParadoxV5) [inline mysql-test/suite/rpl/t/rpl_master_retry_count.test:43]: #### (this thing’s definitely not on)

```suggestion
    CHANGE MASTER TO master_connect_retry= master_connect_retry;
```
Also, I forgot to reset this `master_connect_retry` at the end.

**Comment by ParadoxV5** [comment]
Superceded by #4220
